pkgs/applications/version-management/sourcehut/core.nix

```diff
@@ -92,6 +89,13 @@ buildPythonPackage rec {
   dontUseSetuptoolsCheck = true;
   pythonImportsCheck = [ "srht" ];
 
+  # "actual" is a misnomer. We need to pass through these inputs to dependent services but we also don't want
+  # to pass through hooks, because we don't want hooks to run twice (when it's possible for them to do so).
+  passthru.actualNativeBuildInputs = [
+    sassc
+    nodejs
+  ];
```

Comment:
I'm really impressed by the mental leap you took to realize this was the problem.

Have a few suggestions coming from someone who didn't figure this out, which could be more representative of a future reader of this derivation. Nothing mandatory, just opinion.

1. What do you think of calling this `coreSrhtNativeBuildInputs`? This emphasizes that the source of the inputs is the `core.sr.ht` repo. Or, `sharedSrhtNativeBuildInputs`, which speaks to the purpose of the `core.sr.ht` repo as a container for "code shared among all sr.ht projects" (c.f. the [README](https://git.sr.ht/~sircmpwn/core.sr.ht)).
2. Moving this comment to the definition of `nativeBuildInputs` a few lines above can be better. I feel that, aside from explanation, the main impact of the comment is to discourage depending on `nativeBuildInputs` directly, which makes sense when next to the property itself. It could offer a warning to not use it because [insert your explanation of non-idempotent hooks being added more than once].

Reply:
> 1. What do you think of calling this `coreSrhtNativeBuildInputs`? This emphasizes that the source of the inputs is the `core.sr.ht` repo. Or, `sharedSrhtNativeBuildInputs`, which speaks to the purpose of the `core.sr.ht` repo as a container for "code shared among all sr.ht projects" (c.f. the [README](https://git.sr.ht/~sircmpwn/core.sr.ht)).

I like the sound of `sharedNativeBuildInputs` (which in usage would be `srht.sharedNativeBuildInputs`), thanks!

> 2. Moving this comment to the definition of `nativeBuildInputs` a few lines above can be better. I feel that, aside from explanation, the main impact of the comment is to discourage depending on `nativeBuildInputs` directly, which makes sense when next to the property itself. It could offer a warning to not use it because [insert your explanation of non-idempotent hooks being added more than once].

I've moved the comment and reworded it a bit -- let me know what you think. (Is the passthru fine where it is, though? They're so far apart, so not sure.)

Reply:
I think it's fine.